### PR TITLE
Docs Design

### PR DESCRIPTION
**Pull request by avigoldman**

### CODE REVIEW

**Comment by macklinu** [comment]
This is cool to see, thanks for making progress on this @avigoldman

**Comment by avigoldman** [comment]
I'm good with merging this first pass of the docs. 

Not responsive yet, and there are a lot of stubs, but it is an improvement.

File: examples/docs/src/templates/docs.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+      "Helvetica Neue",
+      sans-serif,
+      "Apple Color Emoji",
+      "Segoe UI Emoji",
+      "Segoe UI Symbol";
+
+    font-size: 16px;
+  }
+
+  a {
```
Comment (ChristopherBiscardi) [inline examples/docs/src/templates/docs.js]: global link style seems weird (especially on a tag), should move it into the link component or something like that.

File: examples/docs/src/components/sidebar.js [hunk 2]
```diff
@@ -0,0 +1,203 @@
+import React from "react";
+import { StaticQuery, graphql } from "gatsby";
+import styled from "react-emotion";
+import flatten from "lodash.flatten";
+import { ExternalLink } from "react-feather";
+import Link from "./link";
+
+// prettier-ignore
+const nav = [
```
Comment (ChristopherBiscardi) [inline examples/docs/src/components/sidebar.js]: this whole thing should be hard coded as JSX or programmatically generated. This "JSON Spec" approach is an anti-pattern for non-data content (that is, data that would display in a large table or infinite list, etc).

I'm mostly concerned about having to maintain a manual list of all of the documentation. Adding a file should "just work" when it comes to being in the navigation, displayed on the site, etc. Other than the top level nav, use alphabetical sorting.

File: examples/docs/gatsby-node.js [hunk 3]
```diff
@@ -70,11 +69,16 @@ exports.onCreateNode = ({ node, getNode, actions }) => {
 
   if (node.internal.type === `Mdx`) {
     const parent = getNode(node.parent);
-    const value = parent.relativePath.replace(parent.ext, "");
+    let value = parent.relativePath.replace(parent.ext, "");
+
+    if (value === "getting-started") {
```
Comment (ChristopherBiscardi) [inline examples/docs/gatsby-node.js]: seems... weird? Why is `getting-started` special-cased here?

File: examples/docs/src/templates/docs.js (hunk 1 above)
Comment (ChristopherBiscardi) [reply examples/docs/src/templates/docs.js]: this can be done as future work.

**Comment by ChristopherBiscardi** [comment]
also looks like there's a merge conflict

**Comment by ChristopherBiscardi** [comment]
also, I feel pretty strongly that the front-page should be a marketing "what is this, why do I need it, etc" page, not the getting started doc. Maybe better to remove the custom handling of the `getting-started` page that redirects it to be the `/` path and leave an `<h1>Gatsby+MDX</h1>` with the repo description under it for now.

File: examples/docs/src/components/sidebar.js [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+    &:hover {
+      color: #663399;
+    }
+
+    ${props =>
+      props.stub &&
+      `
+      color: #666;
+      font-weight: 500;
+    `} ${props =>
```
[the remark this answers is not in the document]
Comment (avigoldman) [reply examples/docs/src/components/sidebar.js]: Prettier messing with things.

File: examples/docs/src/components/sidebar.js [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+        const { node } = allMdx.edges.find(
+          ({ node }) => node.fields.slug === navItem
+        );
+
+        return (
+          <ListItem
+            to={`/${node.fields.slug}`}
+            level={level}
+            active={location.pathname === node.fields.slug}
+            stub={node.frontmatter.stub}
```
[the remark this answers is not in the document]
Comment (avigoldman) [reply examples/docs/src/components/sidebar.js]: All for automation!

File: examples/docs/gatsby-node.js (hunk 3 above)
Comment (avigoldman) [reply examples/docs/gatsby-node.js]: For the homepage. I'm good with having a more classical home page.

File: examples/docs/src/templates/docs.js [hunk 6]
```diff
@@ -0,0 +1,107 @@
+import React, { Component } from "react";
+import { graphql } from "gatsby";
+import MDXRenderer from "gatsby-mdx/mdx-renderer";
+import styled, { injectGlobal } from "react-emotion";
+import { Edit3 } from "react-feather";
+import { Layout, Link } from "$components";
+
+injectGlobal`
```
[the remark this answers is not in the document]
Comment (avigoldman) [reply examples/docs/src/templates/docs.js]: I'll move this to the layout component (same way gatsby website handles global css)

**Pull request merged**